src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -138,6 +139,12 @@ const WorkspaceContainer = ({
   const [leavingWorkspace, setLeavingWorkspace] = useState(false)
   const workspaceLoaded = !!workspace
 
+  useEffect(() => {
+    if (workspaceLoaded && isGoogleWorkspace(workspace) && workspace.workspaceInitialized === false) {
```

Comment:
This is still emitting events on every render because it isn't paying attention to when the state changes. I'm confused why you aren't getting a lint error that the dependency isn't declared.

Let me play with this a little and I will push something up.

Reply:
Thank  you so much for the assist!

Reply:
Interestingly I got a lint error when I tried to commit (and still had one undeclared dependency). Did you commit without running the precommit check?

```
/Users/roberts/Documents/source_new/terra-ui/src/pages/workspaces/workspace/WorkspaceContainer.js
  147:6  warning  React Hook useEffect has a missing dependency: 'workspace'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
✖ 1 problem (0 errors, 1 warning)